reasoner_pydantic/kgraph.py

```diff
@@ -109,6 +109,10 @@ def update(self, other):
         if other.sources:
             if self.sources:
                 self.sources.update(other.sources)
```

Comment:
Based on our Teams chat, shouldn't this be removed, as this just smushes the two lists together?

Reply:
We do still need to do union of the two sets, since we need the new set to have everything in the other set that it doesn't already.